### PR TITLE
HARP-13493: Make maxZoomLevel value in technique non inclusive

### PR DESCRIPTION
**Pull request by germanz**
Signed-off-by: German Zargaryan <german.zargaryan@here.com>

Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

File: @here/harp-datasource-protocol/lib/StyleSetEvaluator.ts [hunk 1]
```diff
@@ -693,7 +693,7 @@ export class StyleSetEvaluator {
                 }
             }
 
-            if (typeof maxZoomLevel === "number" && zoomLevel > maxZoomLevel) {
```
Comment (atomicsulfate) [inline @here/harp-datasource-protocol/lib/StyleSetEvaluator.ts:696]: We need to document this change in BaseTechniqueParams in @here/harp-datasource-protocol/lib/TechniqueParams.ts.

**Review by atomicsulfate — COMMENTED** [review]
Please document the behaviour change as part of the next release notes:
https://github.com/heremaps/harp.gl/releases/

There's not draft of next release (0.22) so you can create a new one. Just follow the same format as the previous releases. Once done click on "Save draft", not on "Publish release".

File: @here/harp-datasource-protocol/lib/StyleSetEvaluator.ts (hunk 1 above)
Comment (germanz) [reply @here/harp-datasource-protocol/lib/StyleSetEvaluator.ts:696]: done

**Review by atomicsulfate — DISMISSED** [review]

**Comment by nzjony** [comment]
I agree with the idea of making it non-inclusive, however the naming is unfortunate, the `max` in `maxZoomLevel` is short for "maximum", which (according to google), means: "the greatest amount, extent, or intensity possible, permitted, or recorded.", i.e. the maximum always refers to an attainable value, however this changes breaks this concept.

I think a better name could be start `beginZoomLevel` and `limitZoomLevel`, where "limit" is (at least in mathematics) clear that it is a value that we get close to, but don't reach.

**Comment by germanz** [comment]
retest this please

**Comment by germanz** [comment]
@nzjony I agree with you about the definition of the term. However, I think it is better to keep it simple and document the range `[minZoomLevel, maxZoomLevel)` so that the upper edge is not reachable. 

It will be unfortunate to break the compatibility by renaming the property or introducing another one and cause confusion. Other Map rendering APIs also use the same terms.

File: @here/harp-datasource-protocol/lib/TechniqueParams.ts [hunk 2]
```diff
@@ -264,7 +264,7 @@ export interface BaseTechniqueParams {
     minZoomLevel?: DynamicProperty<number>;
 
     /**
-     * Maximum zoom level. If the current zoom level is larger, the technique will not be used.
+     * Maximum zoom level. If the current zoom level is equal to or greater, the technique will not be used.
```
Comment (robertoraggi) [inline @here/harp-datasource-protocol/lib/TechniqueParams.ts:267]: You may want to check with our Design team before going forward with this change of semantics.

Comment (robertoraggi) [reply @here/harp-datasource-protocol/lib/TechniqueParams.ts:267]: Also, shouldn't you update the semantics of all the zoom/data level constraints? e.g. `DataSource.maxDisplayLevel`, `DataSource.maxDataLevel`, `MapView.maxZoomLevel`, `MapControls.maxZoomLevel`?

**Review by robertoraggi — APPROVED** [review]

File: @here/harp-datasource-protocol/lib/TechniqueParams.ts (hunk 2 above)
Comment (germanz) [reply @here/harp-datasource-protocol/lib/TechniqueParams.ts:267]: As agreed for the time being we leave the semantic of those properties as-is, both edges inclusive.

**Pull request merged**